ubclaunchpad/bounce-front-end
# Added redux - installed packages, written configureStore, testing instructions below.

:tickets: **Ticket(s)**: Closes #115

---

## :construction_worker: Changes

- Installed redux,react-redux,redux-thunk.

- Housekeeping to directory structure.

- Written configureStore.js which serves as a store for all state in the application. Based on the principles of redux, this store is immutable.
-Written some test reducers etc to make sure the store works properly. 

## :traffic_light: Concerns

_If there is anything you want reviewers to pay special attention to, put it here._

## :mag: Testing Instructions

- npm start 

-  Go to the console 
-   `store.getState()` should return the original state which is: `{articles: Array(0)}`.
- Subscribe is a method that accepts a callback that will fire whenever an action is dispatched.
  Dispatching an action basically means notifying the store that we want to change the state. 
`store.subscribe(() => console.log('Redux!!'))`
- To dispatch an action call `store.dispatch( addArticle({ title: 'React Redux is cool', id: 1 }) )`
- After running the above line, console should print: `Redux!!`
- Check the state is updated by calling: `store.getState()`, you should get: `{articles: Array(1)}`.
*Articles is a javascript object I defined arbitrarily.
